### PR TITLE
Get block passing working with comms

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
Provide a brief description of what this PR does, and why it's needed.

Note this PR needs a lot of clean up but can be merged into #623 

Example output

```
$ java -jar target/scala-2.12/rnode-assembly-0.2.1.jar --bootstrap "rnode://87184a84fd9e4ef9825349701feb93ae@159.203.84.29:30304"
...
01:22:12.171 [scala-execution-context-global-25] INFO  logger - Sent block 3ffde56587264a9fdc30c83507d7e2d757603877453576693d41d4ac782fad99 to #{PeerNode 87184a84fd9e4ef9825349701feb93ae}
01:22:12.171 [scala-execution-context-global-25] INFO  logger - Received block 3ffde56587264a9fdc30c83507d7e2d757603877453576693d41d4ac782fad99. New fork-choice is TODO.
01:22:12.173 [scala-execution-context-global-25] INFO  logger - Sent block 714fbcf2e84da8ca2677bafde9309d2fb46e5e76fa9b09771e57bcabf9022323 to #{PeerNode 87184a84fd9e4ef9825349701feb93ae}
01:22:12.173 [scala-execution-context-global-25] INFO  logger - Received block 714fbcf2e84da8ca2677bafde9309d2fb46e5e76fa9b09771e57bcabf9022323. New fork-choice is TODO.
01:22:12.179 [scala-execution-context-global-25] INFO  logger - Sent block 4c333d10fae134670cec5b86dd259a023b37957a4abe034d19e39678ce453aab to #{PeerNode 87184a84fd9e4ef9825349701feb93ae}
01:22:12.179 [scala-execution-context-global-25] INFO  logger - Received block 4c333d10fae134670cec5b86dd259a023b37957a4abe034d19e39678ce453aab. New fork-choice is TODO.
01:22:12.184 [scala-execution-context-global-23] INFO  logger - Sent block 38ec36de6bd8e7ba20a7e9ca500de12a14ad0fd5cb962e3b22abd9e56214c708 to #{PeerNode 87184a84fd9e4ef9825349701feb93ae}
01:22:12.184 [scala-execution-context-global-23] INFO  logger - Received block 38ec36de6bd8e7ba20a7e9ca500de12a14ad0fd5cb962e3b22abd9e56214c708. New fork-choice is TODO.
01:22:12.188 [scala-execution-context-global-23] INFO  logger - Sent block 2deb2f8c613ba35b539c0bc1a9a216b46c0675cf809ce6d0ff34ef2ee45ef828 to #{PeerNode 87184a84fd9e4ef9825349701feb93ae}
01:22:12.189 [scala-execution-context-global-23] INFO  logger - Received block 2deb2f8c613ba35b539c0bc1a9a216b46c0675cf809ce6d0ff34ef2ee45ef828. New fork-choice is TODO.
Goodbye.
```

### Does this PR relate to an RChain JIRA issue? 
If applicable, add link to corresponding JIRA issue.

This is for the Node v0.3 release.

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/util/comm/CommUtil.scala [hunk 1]
```diff
@@ -2,61 +2,62 @@ package coop.rchain.casper.util.comm
...
+
 import scala.util.Try
 
 object CommUtil {
-  def sendBlock[F[_]: Monad: NodeDiscovery: TransportLayer: Log](b: BlockMessage): F[Unit] = {
-    val packet = blockMessageToPacket(b)
-    val msg    = framePacket(packet)
-
+  def sendBlock[
+      F[_]: Monad: NodeDiscovery: TransportLayer: Log: Time: Encryption: KeysStore: ErrorHandler](
```
Comment (goral09) [inline casper/src/main/scala/coop/rchain/casper/util/comm/CommUtil.scala]: Looking at all this type classes required for `F[_]` -- doesn't it break encapsulation principle? Why caller of this should ever by worried by all of it? I should be able to call `sendBlock` with arguments and that's it. I don't care if it's using encryption, whether it logs errors or measure execution time or that it need some `TransportLayer`. I think I prefer pure OO with proper encapsulation than this. Any thoughts? ping @rabbitonweb @henrytill

Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/util/comm/CommUtil.scala]: I think the issue is that we have to call frameMessage[F], which ends up pulling all those type classes in. I think TransportLayer[F].commSend should just be able to take a Packet .

File: casper/src/main/scala/coop/rchain/casper/Casper.scala [hunk 2]
```diff
@@ -19,12 +27,39 @@ object MultiParentCasper extends MultiParentCasperInstances {
...
+  def simpleCasper[
+      F[_]: Monad: Capture: Applicative: NodeDiscovery: TransportLayer: Log: Time: Encryption: KeysStore: ErrorHandler]
+    : MultiParentCasper[F] = new MultiParentCasper[F] {
     def addBlock(b: BlockMessage): F[Unit]    = ().pure[F]
     def contains(b: BlockMessage): F[Boolean] = false.pure[F]
     def deploy(r: Resource): F[Unit]          = ().pure[F]
     def estimator: F[IndexedSeq[BlockMessage]] =
       Applicative[F].pure[IndexedSeq[BlockMessage]](Vector(BlockMessage()))
     def proposeBlock: F[Option[BlockMessage]] = Applicative[F].pure[Option[BlockMessage]](None)
+    def shouldSendBlock: F[Unit] =
```
Comment (rabbitonweb) [inline casper/src/main/scala/coop/rchain/casper/Casper.scala]: maybe rename that to `sendBlockWhenReady` ?

File: casper/src/main/scala/coop/rchain/casper/util/comm/CommUtil.scala (hunk 1 above)
Comment (rabbitonweb) [reply casper/src/main/scala/coop/rchain/casper/util/comm/CommUtil.scala]: @goral09 I'm more then happy to have a conversation about it today, after standup?

**Pull request merged**